app/src/main/kotlin/com/wire/android/ui/userprofile/UserProfileScreen.kt

```diff
@@ -31,68 +34,129 @@ import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
 import com.wire.android.R
 import com.wire.android.model.UserStatus
 import com.wire.android.ui.common.Icon
 import com.wire.android.ui.common.UserProfileAvatar
 import com.wire.android.ui.common.UserStatusIndicator
+import com.wire.android.ui.common.WireDialog
+import com.wire.android.ui.common.WireDialogButtonProperties
+import com.wire.android.ui.common.WireDialogButtonType
 import com.wire.android.ui.common.button.WireButtonState
 import com.wire.android.ui.common.button.WireSecondaryButton
 import com.wire.android.ui.common.dimensions
 import com.wire.android.ui.common.selectableBackground
 import com.wire.android.ui.common.textfield.WirePrimaryButton
 import com.wire.android.ui.theme.wireTypography
+import com.wire.android.ui.userprofile.UserProfileNavigation.ProfileImage
+import com.wire.android.ui.userprofile.UserProfileNavigation.UserProfile
+import com.wire.android.ui.userprofile.image.ProfileImageScreen
+import io.github.esentsov.PackagePrivate
 
-@OptIn(ExperimentalMaterial3Api::class)
+
+@OptIn(ExperimentalMaterialApi::class, ExperimentalMaterial3Api::class)
 @Composable
-fun UserProfileScreen(viewModel: UserProfileViewModel) {
-    with(viewModel.userProfileState) {
-        UserProfileScreen(
-            state = this,
-            onCloseClick = { viewModel.close() },
-            onLogoutClick = { viewModel.logout() },
-            onEditClick = { viewModel.editProfile() },
-            onStatusClicked = { viewModel.changeStatusClick(it) },
-            onAddAccountClick = { viewModel.addAccount() },
-            dismissStatusDialog = { viewModel.dismissStatusDialog() },
-            onStatusChange = { viewModel.changeStatus(it) },
-            onNotShowRationaleAgainChange = { show -> viewModel.dialogCheckBoxStateChanged(show) }
+fun UserProfileRoute(viewModel: UserProfileViewModel) {
+    val navHostController = rememberNavController()
+
+    NavHost(
+        navController = navHostController,
+        startDestination = UserProfile.route
```

Comment:
Because as I am aware we do not use this anywhere else in the app also using it as a main Navigation i think would screw the back button behaviour or we will need to do some crazy stuff with Navigation manually as I am aware, also the userprofilescreen should have a longer lifecycle i think to correctly pass the bitmap which i will submit in second PR so maybe we can look into it again on my second PR?

Reply:
Image picker will be used in ConversationScreen I believe.
and about back button: nothing screwed, it will be just like `SettingsScreen`: back button navigates back

Reply:
Image picker is a seperate componet that can be reused. I will try to apply your sugestion in a second PR and see if it improves anything